pkg/user/tx_client.go

```diff
@@ -566,6 +575,35 @@ func (client *TxClient) handleEvictions(txHash string) error {
 	return fmt.Errorf("tx was evicted from the mempool")
 }
 
+// isTransactionCommitted checks if a transaction has been committed to the blockchain
+// by querying the node directly. This helps distinguish between true evictions and
+// false positives where a tx was removed from mempool but was included in a block by another node's mempool.
+func (client *TxClient) isTransactionCommitted(txHash string) (bool, error) {
+	serviceClient := sdktx.NewServiceClient(client.conns[0])
+
+	maxRetries := 10
+	retryDelay := 500 * time.Millisecond
+
+	for i := 0; i < maxRetries; i++ {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+
+		_, err := serviceClient.GetTx(ctx, &sdktx.GetTxRequest{Hash: txHash})
+		cancel()
+
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") || strings.Contains(err.Error(), "tx does not exist") {
+				time.Sleep(retryDelay)
+				continue
+			}
+			// The tx wasn't found and it isn't a "not found" error.
+			return false, err
+		}
+		// The tx was found, so it was committed
+		return true, nil
+	}
+	return false, nil
+}
+
```

Comment:
I'd like to avoid nested if statements especially in production code if possible

Reply:
i hear you